### DISCUSSION TITLE
修改之后无法提交

### DISCUSSION THREAD

**Discussion by Jeffzholy** [reactjs/zh-hans.react.dev#1240] [General]
- 问题：翻译，修改完毕之后无法提交，如下内容所示。
- 操作步骤：1. 拉取最新内容 2. yarn 安装 3. 翻译文本 4. 提交代码失败
- 可能问题定位：1. 运行 `fix-headings` 脚本失败退出，2. 翻译文件内容可能全文没有 `heading` ，3. 尝试修改有 `heading` 的文件，一样失败退出，4. 网络上搜索解决办法，没有获取到有用的信息
- 现求助社区，希望小伙伴们帮忙

```bash
    ❯ src/**/*.md — 1 file
      ✖ yarn fix-headings [FAILED]
↓ Skipped because of errors from tasks. [SKIPPED]
✔ Reverting to original state because of errors...
✔ Cleaning up temporary files...

✖ yarn fix-headings:
Internal Error: react-dev@workspace:.: This package doesn't seem to be present in your lockfile; try to make an install to update your resolutions
    at J.getCandidates (/usr/local/lib/node_modules/yarn/bin/yarn.js:2:355536)
    at n.getCandidates (/usr/local/lib/node_modules/yarn/bin/yarn.js:2:344526)
    at /usr/local/lib/node_modules/yarn/bin/yarn.js:2:366182
    at Module.w (/usr/local/lib/node_modules/yarn/bin/yarn.js:2:425653)
    at C (/usr/local/lib/node_modules/yarn/bin/yarn.js:2:366145)
error Command failed with exit code 1.
```

**Comment by Yucohny** [comment]
如果仅仅是翻译文档，可以不用 `yarn` 安装依赖运行。翻译文本后直接 `commit` 提交然后 `push` 到 fork 后的仓库即可。

**Comment by Jeffzholy** [comment]
@Yucohny 明白，只是想清楚问题来源，否则提交时还要绕开 `lint-staged` 检查。我再保持这个 issue 开放几天到周末，这两天如果有跟进会在这里继续讨论

**Comment by awxiaoxian2020** [comment]
> 如果仅仅是翻译文档，可以不用 `yarn` 安装依赖运行。翻译文本后直接 `commit` 提交然后 `push` 到 fork 后的仓库即可。

我们的仓库有 husky 限制。

**Reply by Jeffzholy** [reply]
husky 应该只是给在 git  的特定阶段进行某些任务的执行，具体到这个 repo 是执行 `lint-staged`，`lint-staged` 执行的是 `fix-headings` 命令，失败是因为 `scripts/headingIdLinter.js` 这个脚本的执行没通过。我的问题其实就是，为什么这个会失败。

**Comment by awxiaoxian2020** [comment]
`yarn check-all` 请尝试使用该命令进行定位问题文件位置。如果无法定位，请使用 `git commit -m "xxx" --no-verify` 进行强制提交，发起 PR，由维护者协助检查。

**Reply by Jeffzholy** [reply]
这个命令是对 React 组件的 lint 和 format，提交失败应该是脚本检查 `markdown` 文件失败的

**Reply by awxiaoxian2020** [reply]
那先提交上来看看吧

**Comment by Jeffzholy** [comment]
~~记录下，方便后来者查看参考：查看文件 `scripts/headingIDHelpers/generateHeadingIDs.js` 的第 87-91  行，推测有可能是因为在 common module 中使用 `import`  语法导致的执行错误，查看[英文网站对应文件](https://github.com/reactjs/react.dev/blob/main/scripts/headingIDHelpers/generateHeadingIDs.js) 也是一样的。~~
